app/assets/v2/js/pages/faucet_form.js

```diff
@@ -0,0 +1,69 @@
+ $('document').ready(function () {
+    $('#comment').bind('input propertychange', function() {
+      this.value = this.value.replace(/ +(?= )/g,'');
+      if(this.value.length > 500) {
+        this.value = this.value.substring(0, 500);
+      }
+      if(this.value.length){
+        $("#charcount").html(501 - this.value.length);
+      }
+    });
+    $('#githubProfile').on('change', function() {
+      var cleanedUser = this.value.replace('@','').toLowerCase();
+      request_url = '/api/v0.1/faucet/githubProfile/' + cleanedUser;
+      document.getElementById("submitFaucet").setAttribute("disabled","disabled");
+      $.get(request_url, function(result){
+        result = sanitizeAPIResults(result);
+        if(result.user === false) {
+          $('#githubProfile').addClass('is-invalid');
+          $('#githubProfileHelpBlock').html('We could not find that user in Github').show();
+        } else if(result.name === cleanedUser) {
+          $('#githubProfile').addClass('is-invalid-error');
+          $('#githubProfileHelpBlock').html('We have a pending or processed faucet contribution for that Github user').show();
+        }
+        document.getElementById("submitFaucet").removeAttribute("disabled");
```

Comment:
why the transition back and forth from jquery to plain vanilla JS?

Reply:
no good reason.... habit... I prefer vanilla js because I am a control freak :-)